src/internal.c

```diff
@@ -1220,7 +1221,7 @@ int GenerateKey(byte hashId, byte keyId,
         return WS_BAD_ARGUMENT;
     }
 
-    if (k[0] & 0x80) kPad = 1;
+    if (doKeyPad && (k[0] & 0x80)) kPad = 1;
```

Comment:
Lets add a comment here on why we are not doing the padding. Something citing "Thus, K_PQ and K_CL are encoded as fixed-length byte arrays, not as integers. Byte arrays are defined in Section 5 of [[RFC4251](https://www.ietf.org/id/draft-kampanakis-curdle-ssh-pq-ke-01.html#RFC4251)]." Which is why I currently am assuming that we are skipping the padding here in special cases.